### PR TITLE
codeintel: Add "prioritize index" button

### PR DESCRIPTION
**Pull request by efritz**
Add a button to send a precise index to the front of the queue. Right now this just works by modifying the uploaded_at/queued_at timestamp to be a year in the past. This might give a quick win for demoability on dotcom, and something easy to remove from the upcoming branch cut.

## Test plan

Tested locally, added unit tests to store.

### CODE REVIEW

File: enterprise/internal/codeintel/uploads/internal/store/indexes.go [hunk 1]
```diff
@@ -302,6 +302,24 @@ const deleteIndexByIDQuery = `
 DELETE FROM lsif_indexes WHERE id = %s RETURNING repository_id
 `
 
+// TODO -  test
```
Comment (efritz) [inline enterprise/internal/codeintel/uploads/internal/store/indexes.go:305]: TODO

File: enterprise/internal/codeintel/uploads/internal/store/uploads.go [hunk 2]
```diff
@@ -804,6 +804,24 @@ WHERE id = %s
 RETURNING repository_id
 `
 
+// TODO - test
```
Comment (efritz) [inline enterprise/internal/codeintel/uploads/internal/store/uploads.go:807]: TODO

File: enterprise/internal/codeintel/uploads/internal/store/indexes.go [hunk 3]
```diff
@@ -302,6 +302,24 @@ const deleteIndexByIDQuery = `
...
+	}})
+	defer endObservation(1, observation.Args{})
+
+	_, prioritized, err := basestore.ScanFirstInt(s.db.Query(ctx, sqlf.Sprintf(prioritizeIndexByIDQuery, id)))
+	return prioritized, err
+}
+
+const prioritizeIndexByIDQuery = `
+UPDATE lsif_indexes u
+SET queued_at = NOW() - '1 year'::interval, process_after = NULL
```
[the remark this answers is not in the document]
Comment (keynmol) [reply enterprise/internal/codeintel/uploads/internal/store/indexes.go:318]: Very ambitious to think that we won't have indexes in there older than that :trollface:

[the remark this answers is not in the document]
Comment (efritz) [reply enterprise/internal/codeintel/uploads/internal/store/indexes.go:318]: I was going to say that but Olaf was in the thread and he'd be too pragmatic for that to be a point that holds any water :laff:

**Pull request closed without merging**